File: Philip/Level0/distiction-of-finite-decimal.py
def solution(a, b):
    answer = 1
    i = 2
    while (i <= a and i <= b) :
        if(a % i == 0 and b % i == 0) :
            a = a / i
            b = b / i
        else :
            i = i + 1
    j = 2
    while(b >= j) :
        if (b % j == 0):
            if(j == 2 or j == 5):
                b = b / j
            else :
                answer = 2
                break
        else:
            j = j + 1
    return answer

File: Tom/Level0/distiction-of-finite-decimal.cpp
using namespace std;

int solution(int a, int b) {
    int answer = 0;
    
    while(b % 2 == 0) b /= 2;
    
    while(b % 5 == 0) b /= 5;
    
    return (a % b == 0) ? 1 : 2;
}
